timer/timer.go

```diff
@@ -0,0 +1,120 @@
+package timer
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/nivista/steady/.gen/protos/common"
+	"github.com/nivista/steady/internal/.gen/protos/messaging"
+)
+
+type (
+	// Timer represents the configuration for the execution of a recurring task.
+	Timer struct {
+		executer  executer
+		scheduler scheduler
+	}
+
+	// Progress records the progress of a recurring task.
+	Progress struct {
+		// CompletedExecutions indicated the number of successfully completed executions.
+		CompletedExecutions int
+
+		// LastExecution indicates which execution was recorded last.
+		LastExecution time.Time
+	}
+
+	executer interface {
+		execute()
+	}
+
+	scheduler interface {
+		schedule(prog Progress, now time.Time) (nextFire time.Time, done bool)
+	}
+)
+
+func getCanceller() (cancelled <-chan struct{}, cancel func()) {
+	var stopped int32
+
+	ch := make(chan struct{})
+
+	cancel = func() {
+		if atomic.CompareAndSwapInt32(&stopped, 0, 1) {
+			close(ch)
+		}
+	}
+
+	return ch, cancel
```

Comment:
Here you have named returns in the method definition but you name the return something else when you actually return. In my opinion you should not have named returns in any method definition. But if you want to use them then lets use them consistently.

Reply:
Ok i'll omit named returns and just describe the return values in a comment.